tests/uint.rs

```diff
@@ -18,38 +19,50 @@ fn assert_serialize(number: u32) {
 }
 
 #[test]
-fn deserialize_uint() {
-    assert_eq!(deserialize_uint_from(100).unwrap(), uint!(100));
-    assert_eq!(deserialize_uint_from(0).unwrap(), uint!(0));
-    assert_eq!(deserialize_uint_from(9007199254740991i64).unwrap(), UInt::MAX);
-    assert!(deserialize_uint_from(9007199254740992i64).is_err());
+fn deserialize() {
+    assert_eq!(deserialize_from(100).unwrap(), uint!(100));
+    assert_eq!(deserialize_from(0).unwrap(), uint!(0));
+    assert_eq!(deserialize_from(9007199254740991i64).unwrap(), UInt::MAX);
+    assert!(deserialize_from(9007199254740992i64).is_err());
+}
+
+#[test]
+#[cfg_attr(feature = "float_deserialize", ignore)]
+fn dont_deserialize_integral_float() {
+    assert!(deserialize_from(1.0).is_err());
+    assert!(deserialize_from(9007199254740991.0).is_err());
+    assert!(deserialize_from(9007199254740992.0).is_err());
 }
 
 #[test]
 #[cfg_attr(feature = "lax_deserialize", ignore)]
-fn strict_deserialize_uint() {
-    assert!(deserialize_uint_from(0.5).is_err());
-    assert!(deserialize_uint_from(1.0).is_err());
-    assert!(deserialize_uint_from(9007199254740991.0).is_err());
-    assert!(deserialize_uint_from(9007199254740991.49).is_err());
-    assert!(deserialize_uint_from(9007199254740992.0).is_err());
+fn dont_deserialize_fractional_float() {
+    assert!(deserialize_from(0.5).is_err());
+    assert!(deserialize_from(42.1337).is_err());
```

Comment:
I could also add something like 2^52 - 0.5, that would be a fractional double that uses it's entire mantissa if I am not mistaken.

Reply:
Feel free to add whatever tests you feel are worthwhile.

Reply:
Then I'm going to leave this out since I don't think there's anything fundamentally different from the existing numbers.